### PR TITLE
Updated X12 Realtime Reference Adapter to display inbound request messag...

### PR DESCRIPTION
**Pull request by sailajaa**
...es for the elements

### CODE REVIEW

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 1]
```diff
@@ -36,7 +38,19 @@
  */
 public class AdapterCORE_X12DSRealTimeOrchImpl {
 
+    Log log = LogFactory.getLog(AdapterCORE_X12DSRealTimeOrchImpl.class);
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: should be log4j, and should be private static final.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 2]
```diff
@@ -36,7 +38,19 @@
...
+        log.info("CORE Paylod Type = " + msg.getPayloadType());
+        log.info("CORE Processing Mode = " + msg.getProcessingMode());
+        log.info("CORE Rule version = " + msg.getCORERuleVersion());
+        log.info("CORE Payload Id = " + msg.getPayloadID());
+        log.info("CORE Receiver Id = " + msg.getReceiverID());
+        log.info("CORE Sender Id = " + msg.getSenderID());
+        log.info("CORE TimeStamp = " + msg.getTimeStamp());
+        log.info("CORE Payload = " + msg.getPayload());
+        log.info("End AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
         return new COREEnvelopeRealTimeResponse();
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: You need to build a response.  Need to add a payload and all the other metadata to the response.  See the specs for guidance on dummy data.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 3]
```diff
@@ -36,7 +38,19 @@
...
     public COREEnvelopeRealTimeResponse realTimeRequest(COREEnvelopeRealTimeRequest msg, AssertionType assertion) {
+        log.info("Begin AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+        log.info("CORE Paylod Type = " + msg.getPayloadType());
+        log.info("CORE Processing Mode = " + msg.getProcessingMode());
+        log.info("CORE Rule version = " + msg.getCORERuleVersion());
+        log.info("CORE Payload Id = " + msg.getPayloadID());
+        log.info("CORE Receiver Id = " + msg.getReceiverID());
+        log.info("CORE Sender Id = " + msg.getSenderID());
+        log.info("CORE TimeStamp = " + msg.getTimeStamp());
+        log.info("CORE Payload = " + msg.getPayload());
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: Convert the payload from base64 to human readable string before logging.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 4]
```diff
@@ -36,7 +38,19 @@
  */
 public class AdapterCORE_X12DSRealTimeOrchImpl {
 
+    Log log = LogFactory.getLog(AdapterCORE_X12DSRealTimeOrchImpl.class);
+
     public COREEnvelopeRealTimeResponse realTimeRequest(COREEnvelopeRealTimeRequest msg, AssertionType assertion) {
+        log.info("Begin AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: Begin... log should be trace level.  All the others should be debug.

Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: Should add a null check for the msg before logging.

Comment (nareshcgi) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: Reviewed the code, will verify it again once you address the comments from Jason.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 5]
```diff
@@ -36,7 +38,38 @@
...
+            LOG.info("CORE Receiver Id = " + coreEnvelopeRealTimeRequest.getReceiverID());
+            oResponse.setCORERuleVersion("2.2.0");
+            LOG.info("CORE Rule version = " + coreEnvelopeRealTimeRequest.getCORERuleVersion());
+            oResponse.setPayload("PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4NCjxDbGluaWNhbERvY3VtZW50\n"
+                + "IG1vb2RDb2RlPSJFVk4iIHhtbG5zPSJ1cm46aGw3LW9yZzp2MyINCiAgICB4bWxuczp4c2k9Imh0");
+            LOG.info("CORE Payload = " + Base64Coder.decodeString(coreEnvelopeRealTimeRequest.getPayload()));
+
+            LOG.trace("End AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+
+        }
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java:62]: Add else to this and create an error response or just return an empty response and add a TODO for error handling.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 6]
```diff
@@ -36,7 +38,38 @@
...
+    private static final Logger LOG = Logger.getLogger(AdapterCORE_X12DSRealTimeOrchImpl.class);
+
+    public COREEnvelopeRealTimeResponse realTimeRequest(COREEnvelopeRealTimeRequest coreEnvelopeRealTimeRequest, AssertionType assertion) {
+
+        COREEnvelopeRealTimeResponse oResponse = null;
+        if (coreEnvelopeRealTimeRequest != null) {
+
+            LOG.trace("Begin AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+
+            oResponse = new COREEnvelopeRealTimeResponse();
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: I'd create two private methods, one for logging inbound and one for creating response.  would make the code cleaner and easier to read.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 7]
```diff
@@ -36,7 +38,60 @@
...
+            createAdapterCORE_X12DSRealTimeRequestLogInfo(coreEnvelopeRealTimeRequest);
+            LOG.trace("End AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+        } else {
+            return oResponse;
+            //TODO: Need to add error handling
+        }
+        return oResponse;
+    }
+
+    /**
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: don't need javadocs for private methods.  Should include a description in the public method.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 8]
```diff
@@ -36,7 +38,60 @@
...
+        oResponse.setPayload("PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4NCjxDbGluaWNhbERvY3VtZW50\n"
+            + "IG1vb2RDb2RlPSJFVk4iIHhtbG5zPSJ1cm46aGw3LW9yZzp2MyINCiAgICB4bWxuczp4c2k9Imh0");
+        return oResponse;
     }
+
+    /**
+     *
+     * @param coreEnvelopeRealTimeRequest
+     */
+    private void createAdapterCORE_X12DSRealTimeRequestLogInfo(COREEnvelopeRealTimeRequest coreEnvelopeRealTimeRequest) {
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: method name is create but not really creating anything.  could just be: logAdapterCORE_X12DSRealTimeRequest(request)

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 9]
```diff
@@ -36,7 +38,54 @@
...
+        COREEnvelopeRealTimeResponse oResponse = null;
+        if (coreEnvelopeRealTimeRequest != null) {
+            LOG.trace("Begin AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+            //Call to a method which builds response metadata and returns response
+            oResponse = buildAdapterCORE_X12DSRealTimeResponseMetadata();
+            //Call for logging inbound
+            LogAdapterCORE_X12DSRealTimeRequest(coreEnvelopeRealTimeRequest);
+            LOG.trace("End AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+        } else {
+            return oResponse;
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: missed this before.  it's still null.  you need to add: oResponse = new COREEnvelopeRealTimeResponse();
you don't need a return statement there, just the assignment.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 10]
```diff
@@ -36,7 +38,54 @@
...
+        oResponse.setTimeStamp("2007-08-30T10:20:34Z");
+        oResponse.setSenderID("PayerB");
+        oResponse.setReceiverID("HospitalA");
+        oResponse.setCORERuleVersion("2.2.0");
+        oResponse.setPayload("PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4NCjxDbGluaWNhbERvY3VtZW50\n"
+            + "IG1vb2RDb2RlPSJFVk4iIHhtbG5zPSJ1cm46aGw3LW9yZzp2MyINCiAgICB4bWxuczp4c2k9Imh0");
+        return oResponse;
     }
+
+    private void LogAdapterCORE_X12DSRealTimeRequest(COREEnvelopeRealTimeRequest coreEnvelopeRealTimeRequest) {
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: don't capitalize method names.

File: Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java [hunk 11]
```diff
@@ -36,7 +38,55 @@
...
+        if (coreEnvelopeRealTimeRequest != null) {
+            LOG.trace("Begin AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+            //Call to a method which builds response metadata and returns response
+            oResponse = buildAdapterCORE_X12DSRealTimeResponseMetadata();
+            //Call for logging inbound
+            logAdapterCORE_X12DSRealTimeRequest(coreEnvelopeRealTimeRequest);
+            LOG.trace("End AdapterCORE_X12DSRealTimeOrchImpl.realTimeRequest()");
+        } else {
+            oResponse = new COREEnvelopeRealTimeResponse();
+            return oResponse;
```
Comment (jsmith2) [inline Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/adapter/AdapterCORE_X12DSRealTimeOrchImpl.java]: remove this return statement, it's redundant.

**Pull request merged**